### PR TITLE
feature/ELPP-897 Refactored custom mime type work and updated RAML schema tests.

### PR DESCRIPTION
**Pull request by jamiehollern**
@nlisgo @giorgiosironi

### CODE REVIEW

File: src/modules/jcms_rest/tests/src/PathMimeTypeMapperTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   *   The class we're testing.
+   */
+  protected $pathMimeTypeMapper;
+
+  public function setUp() {
+    $this->pathMimeTypeMapper = new PathMimeTypeMapper();
+  }
+
+  public function dataProvider()
+  {
```
Comment (giorgiosironi) [inline src/modules/jcms_rest/tests/src/PathMimeTypeMapperTest.php]: very useful examples

Comment (jamiehollern) [inline src/modules/jcms_rest/tests/src/PathMimeTypeMapperTest.php]: I've just realised I need to add some more dynamic examples. Thanks, @giorgiosironi!

File: src/modules/jcms_rest/src/Plugin/rest/resource/SubjectsItemRestResource.php [hunk 2]
```diff
@@ -78,6 +79,9 @@ public function get($id = NULL) {
         $response['impactStatement'] = $term->get('field_impact_statement')->first()->getValue()['value'];
       }
 
+      $new_response = new JsonResponse($response, Response::HTTP_OK, ['Content-Type' => 'application/vnd.elife.subject+json;version=1']);
+      return $new_response;
+
       $resource_response = new ResourceResponse($response, $status);
```
Comment (nlisgo) [inline src/modules/jcms_rest/src/Plugin/rest/resource/SubjectsItemRestResource.php]: We can drop these lines now?

File: src/modules/jcms_rest/src/Plugin/rest/resource/SubjectsRestResource.php [hunk 3]
```diff
@@ -54,6 +56,8 @@ public function get() {
         }
       }
     }
+    $new_response = new JsonResponse($response_data, Response::HTTP_OK, ['Content-Type' => 'application/vnd.elife.subject-list+json;version=1']);
+    return $new_response;
     $response = new ResourceResponse($response_data);
     // @todo - Digirati - jamiehollern - Alter response content type header to include the API version.
```
Comment (nlisgo) [inline src/modules/jcms_rest/src/Plugin/rest/resource/SubjectsRestResource.php]: This stuff is redundant now?

**Pull request merged**